src/modal.scss

```diff
@@ -4,111 +4,190 @@
-  $fd-modal-padding-y: fd-space(4) !default;
-  $fd-modal-inner-content-background: fd-color("background", 2) !default;
+  &__content {
+    @include fd-reset();
 
-  @include fd-reset();
+    @include media-breakpoint-down(l) {
+      margin: $fd-modal-content-margin-y-mobile $fd-modal-content-margin-x-mobile;
+    }
 
-  max-width: $fd-modal-width;
+    overflow: hidden;
+    min-width: $fd-modal-content-min-width;
+    min-height: $fd-modal-content-min-height;
+    box-shadow: $fd-modal-content-box-shadow;
+    border-radius: $fd-modal-content-border-radius;
+    margin: $fd-modal-content-margin-y $fd-modal-content-margin-x;
+
+    @each $size-label, $size in $fd-modal-content-padding-x {
+      &--#{$size-label} {
+        .#{$block}__header,
+        .#{$block}__body,
+        .#{$block}__footer {
+          padding-left: map_get($fd-modal-content-padding-x, $size-label);
+          padding-right: map_get($fd-modal-content-padding-x, $size-label);
+        }
+      }
+    }
 
-  &__content {
-    @include fd-reset();
+    &--compact {
+      .#{$block}__header {
+        height: $fd-modal-header-compact-height;
+      }
 
-    border-radius: $fd-border-radius;
+      .#{$block}__footer {
+        height: $fd-modal-footer-compact-height;
+      }
+    }
 
-    @include fd-var-color("background-color", $fd-modal-inner-content-background, --fd-modal-background-color);
+    &--resizable {
+      display: flex;
+      flex-direction: column;
+      position: relative;
+
+      .#{$block}__body {
+        flex-grow: 1;
+      }
+
+      .#{$block}__header,
+      .#{$block}__footer {
+        flex-shrink: 0;
+      }
+
+      @include media-breakpoint-up(l) {
+        resize: both;
+
+        @include fd-icon("resize-corner", "m", "after") {
+          color: $fd-modal-resize-handle-color;
+          position: absolute;
+          right: 0;
+          bottom: 0;
+        }
+      }
+    }
   }
 
   &__header,
   &__body,
   &__footer {
-    @include fd-reset();
-    @include fd-var-size("padding-left", $fd-modal-padding-x, --fd-modal-padding-x);
-    @include fd-var-size("padding-right", $fd-modal-padding-x, --fd-modal-padding-x);
-
-    padding-top: $fd-modal-padding-y;
-    padding-bottom: $fd-modal-padding-y;
+    padding-left: map_get($fd-modal-content-padding-x, "def");
+    padding-right: map_get($fd-modal-content-padding-x, "def");
   }
 
   &__header {
     position: relative;
-    border-bottom-style: solid;
-    border-bottom-width: 1px;
-
-    @include fd-var-color("border-bottom-color", $fd-modal-border-color, --fd-modal-border-color);
-  }
-
-  &__title {
-    @include fd-heading();
-    @include fd-type("1");
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    height: $fd-modal-header-height;
+    box-shadow: $fd-modal-header-box-shadow;
+    background-color: $fd-modal-header-color;
+
+    .#{$block}__title {
+      @include fd-reset();
+      @include fd-type("1");
```

Comment:
Should get rid of `fd-type`, it's a hard-coded value and won't be affected by the theme. You can use css variable from sap-theming